src/core/coap/coap_message.cpp

```diff
@@ -502,5 +389,140 @@ const char *Message::CodeToString(void) const
 }
 #endif // OPENTHREAD_CONFIG_COAP_API_ENABLE
 
+otError OptionIterator::Init(const Message *aMessage)
+{
+    otError err = OT_ERROR_NONE;
+
+    /*
+     * Check that:
+     *   Length - Offset: the length of the payload
+     * is greater than
+     *   Start position of optionsa
+     *
+     * → Assert options start before the message ends
```

Comment:
Do we really assert?

Reply:
Maybe "assert" is the wrong verb (it's a "soft assert": if condition isn't met we bail the `Init` function with `OT_ERROR_PARSE` as the reason).

I put the comment there because it took me a few minutes to decipher what the `VerifyOrExit` check was doing and what it would do in the "no options" case (i.e. did I need to make `>=` a `>`, check for `==` separately, etc).